File: JAVA/src/daysOfCode30/QueuesAndStacks.java
package daysOfCode30; 

import java.util.*;
import java.util.LinkedList;


public class QueuesAndStacks {
    // Write your code here.
    Stack pila = new Stack();
    LinkedList cola = new LinkedList();
    
    //Pushes character onto a stack.
    public void pushCharacter(char ch){
        pila.push(ch);
    }
    
    //Enqueues a character in the queue instance variable
    public void enqueueCharacter(char ch){
    	cola.addLast(ch);
    }
    
    //Pops and returns the character at the top of the stack instance variable    
    public char popCharacter(){
    	return (char)pila.pop();
    }
    
    //Dequeues and returns the first character in the queue instance variable.
    public char dequeueCharacter(){
    	return (char)cola.removeFirst();
    }
    
    
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String input = scan.nextLine();
        scan.close();

        // Convert input String to an array of characters:
        char[] s = input.toCharArray();

        // Create a Solution object:
        QueuesAndStacks p = new QueuesAndStacks();

        // Enqueue/Push all chars to their respective data structures:
        for (char c : s) {
            p.pushCharacter(c);
            p.enqueueCharacter(c);
        }

        // Pop/Dequeue the chars at the head of both data structures and compare them:
        boolean isPalindrome = true;
        for (int i = 0; i < s.length/2; i++) {
            if (p.popCharacter() != p.dequeueCharacter()) {
                isPalindrome = false;                
                break;
            }
        }

        //Finally, print whether string s is palindrome or not.
        System.out.println( "The word, " + input + ", is " 
                           + ( (!isPalindrome) ? "not a palindrome." : "a palindrome." ) );
    }
    
    
}

File: PHP/30DaysOfCode/day18/QueuesAndStacks.php
<?php

class Solution {
    // Write your code here
    protected $pila = array();
    protected $cola = array();
    
    //Pushes character onto a stack.
    public function pushCharacter($ch){
        if($ch !="\n") {
            array_push($this->pila, $ch);
        }
    }
    
    //Enqueues a character in the queue instance variable
    public function enqueueCharacter($ch){
        if($ch != "\n"){
            array_push($this->cola, $ch);
        }
    }
    
    //Pops and returns the character at the top of the stack instance variable 
    public function popCharacter(){
        return array_shift($this->pila);
    }
    
    //Dequeues and returns the first character in the queue instance variable.
    public function dequeueCharacter(){
        return array_pop($this->cola);
    }
}

// read the string s
$s = fgets(STDIN);

// create the Solution class object p
$obj = new Solution();

$len = strlen($s);
$isPalindrome = true;

// push/enqueue all the characters of string s to stack
for ($i = 0; $i < $len; $i++) {
    $obj->pushCharacter($s{$i});
    $obj->enqueueCharacter($s{$i});
}

/*
pop the top character from stack
dequeue the first character from queue
compare both the characters
*/
for ($i = 0; $i < $len / 2; $i++) {
    if($obj->popCharacter() != $obj->dequeueCharacter()){
        $isPalindrome = false;
    	
        break;
    }
}

//finally print whether string s is palindrome or not.
if ($isPalindrome)
    echo "The word, ".$s.", is a palindrome.";
else
    echo "The word, ".$s.", is not a palindrome.";
?>